src/main/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializer.java

```diff
@@ -224,16 +265,62 @@ public DecodedResult deserialize(
           .builder()
           .value(deserializedData)
           .build();
-    } catch (RestClientException e) {
-      cacheSchemaFetchError(e, schemaId);
-      return new DecodedResult(onFailure(encoderOnFailure, bytes), e.getMessage());
-    } catch (IOException | SerializationException e) {
-      var data = onFailure(encoderOnFailure, bytes);
-      Log.error("Error deserializing: %s", data, e);
-      return new DecodedResult(data, e.getMessage());
+    } catch (Exception e) {
+      var exc = e.getCause();
+      if (exc instanceof RestClientException || isNetworkRelatedException(exc)) {
+        // IMPORTANT: We must cache RestClientException, and more importantly,
+        //            network-related IOExceptions, to prevent the sidecar from
+        //            bombarding the Schema Registry servers with requests for every
+        //            consumed message when we encounter a schema fetch error.
+        cacheSchemaFetchError(e, schemaId);
```

Comment:
Yes indeed, I guess it worked anyway because we're keying off the error message, while probably bubbles up from the wrapped extension. Thanks for catching that.

Reply:
Made this change